### DISCUSSION TITLE
Issue with WezTerm and Vim Key Repeat

### DISCUSSION THREAD

**Discussion by Koda777** [wezterm/wezterm#4198] [Q&A] [answered]
Hello everyone,

I've been using WezTerm for my terminal emulator and I've noticed an issue while using Vim. When I'm in Vim and I press certain keys for actions like moving up or down (for example, pressing and holding 'j' to move down), I sometimes encounter a problem. If I release the 'j' key and immediately press another key like 'k' to move up, the 'k' key doesn't seem to register properly. The cursor remains in the same position and doesn't move, forcing me to press the 'k' key again to move up.

This behavior can be a bit frustrating, especially when I'm trying to navigate quickly within Vim. Has anyone else experienced a similar issue with WezTerm and Vim key repeat? Is there a known solution or workaround for this?

I'd appreciate any insights or advice on how to resolve this issue. Thank you in advance for your help!

https://github.com/wez/wezterm/assets/72024743/5187e740-c251-4eea-9df8-73ea29303907

**Comment by Koda777** [comment]
And actually, it's not just happening to me in Vim, but also when I type commands in the terminal, etc. However, in other cases, unlike in Vim, I'm not holding down one key and then immediately switching to another.

**Comment by wez** [comment]
Please see: https://wezfurlong.org/wezterm/troubleshooting.html#debugging-keyboard-related-issues

**Comment by Koda777** [comment]
I understand what's happening. When I quickly press a key right after releasing another key, it doesn't register in a very rapid timeframe. On the video, you can see my Wezterm configuration on the left and Alacritty on the right. If I try to move down with 'j' and immediately after releasing it, move up with 'k', there's a slight delay where the cursor remains still before moving up. However, on the left in Wezterm, it completely freezes and doesn't move upward 


https://github.com/wez/wezterm/assets/72024743/99f21741-23ff-464e-828f-95934b2e1d30

**Reply by Koda777** [reply]
00:03:09.565  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('j'), modifiers: NONE, leds: (empty), phys_code: Some(J), raw_code: 38, repeat_count: 1, key_is_down: false, handled: Handled(false) } 
00:03:09.565  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('j'), modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Char('j'), modifiers: NONE, leds: (empty), phys_code: Some(J), raw_code: 38, repeat_count: 1, key_is_down: false, handled: Handled(false) }) } 
00:03:09.565  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=Char('j') mods=NONE
00:03:09.791  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('k'), modifiers: NONE, leds: (empty), phys_code: Some(K), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
00:03:09.825  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('k'), modifiers: NONE, leds: (empty), phys_code: Some(K), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
00:03:09.859  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('k'), modifiers: NONE, leds: (empty), phys_code: Some(K), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
00:03:09.892  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('k'), modifiers: NONE, leds: (empty), phys_code: Some(K), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
00:03:09.926  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('k'), modifiers: NONE, leds: (empty), phys_code: Some(K), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
00:03:09.960  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('k'), modifiers: NONE, leds: (empty), phys_code: Some(K), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
00:03:09.994  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Char('k'), modifiers: NONE, leds: (empty), phys_code: Some(K), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) } 


And this is the log file when I try to quickly press 'j' and then 'k'. It appears, I believe, that Wezterm doesn't handle the event. On the screen, I only see a 'j' and a 'k' if I type it in the terminal
See handled: Handled(false)

**Reply by wez** [reply, marked as answer]
did you try `use_ime=false` as suggested in the link I sent you?

**Reply by Koda777** [reply]
God i didnt notice that thank you
